order versions appear broken due to recent fsspec changes

@martindurant I see that `fsspec.asyn` module got rid of `maybe_sync` function, but older versions of `s3fs` still rely on the method which now fails to import. For example `s3fs==0.5.2` now does not seem to work after a clean install. I see that s3fs requirements.txt asked for `fsspec>=0.8.0` and that is now incompatible without an upper boundary.

I encounter another error related to `fsspec` when I run `pytest`
```
scrapy_project/project/utils.py:9: in <module>
    import s3fs
../../.conda/envs/d-scraper/lib/python3.8/site-packages/s3fs/__init__.py:1: in <module>
    from .core import S3FileSystem, S3File
../../.conda/envs/d-scraper/lib/python3.8/site-packages/s3fs/core.py:11: in <module>
    from fsspec.utils import infer_storage_options, tokenize, setup_logging
E   ImportError: cannot import name 'setup_logging' from 'fsspec.utils' (/home/me/.local/lib/python3.8/site-packages/fsspec/utils.py)
================================================================== short test summary info ===================================================================
ERROR  - ImportError: cannot import name 'setup_logging' from 'fsspec.utils' (/home/me/.local/lib/python3.8/site-packages/fsspec/utils.py)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================== 1 error in 1.00s ======================================================================

```

--- a/docs/source/changelog.rst
+++ b/docs/source/changelog.rst
@@ -4,7 +4,7 @@ Changelog
 Version 2021.04.0
 -----------------
 
-- swith to calver and fsspec pin
+- switch to calver and fsspec pin
 - py36 (#462)
 - async fixes (#456, 452)
 
@@ -16,7 +16,7 @@ Version 0.6.0
 - cp to preserve ETAG (#441)
 - CI (#435, #427, #395)
 - 5GB PUT (#425)
-- partical cat (#389)
+- partial cat (#389)
 - direct find (#360)
 
 

--- a/docs/source/changelog.rst
+++ b/docs/source/changelog.rst
@@ -1,6 +1,13 @@
 Changelog
 =========
 
+Version 2021.04.0
+-----------------
+
+- swith to calver and fsspec pin
+- py36 (#462)
+- async fixes (#456, 452)
+
 Version 0.6.0
 -------------
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,2 @@
 aiobotocore>=1.0.1
-fsspec>=0.9.0
+fsspec==2021.04.0

--- a/s3fs/tests/test_s3fs.py
+++ b/s3fs/tests/test_s3fs.py
@@ -2012,3 +2012,11 @@ def test_sync_from_wihin_async(s3):
         assert s3.ls(test_bucket_name)
 
     asyncio.run(f())
+
+
+def test_token_paths(s3):
+    fs, tok, files = fsspec.get_fs_token_paths(
+        "s3://" + test_bucket_name + "/*.csv",
+        storage_options={"client_kwargs": {"endpoint_url": endpoint_uri}}
+    )
+    assert files

--- a/s3fs/tests/test_s3fs.py
+++ b/s3fs/tests/test_s3fs.py
@@ -2017,6 +2017,6 @@ def test_sync_from_wihin_async(s3):
 def test_token_paths(s3):
     fs, tok, files = fsspec.get_fs_token_paths(
         "s3://" + test_bucket_name + "/*.csv",
-        storage_options={"client_kwargs": {"endpoint_url": endpoint_uri}}
+        storage_options={"client_kwargs": {"endpoint_url": endpoint_uri}},
     )
     assert files

